sql/core/src/test/resources/sql-tests/inputs/pgSQL/int8.sql

```diff
@@ -0,0 +1,239 @@
+-- 	FROM INT8_TBL;
+
+-- SELECT '' AS to_char_4, to_char( (q1 * -1), '9999999999999999S'), to_char( (q2 * -1), 'S9999999999999999')
+-- 	FROM INT8_TBL;
+
+-- SELECT '' AS to_char_5,  to_char(q2, 'MI9999999999999999')     FROM INT8_TBL;
+-- SELECT '' AS to_char_6,  to_char(q2, 'FMS9999999999999999')    FROM INT8_TBL;
+-- SELECT '' AS to_char_7,  to_char(q2, 'FM9999999999999999THPR') FROM INT8_TBL;
+-- SELECT '' AS to_char_8,  to_char(q2, 'SG9999999999999999th')   FROM INT8_TBL;
+-- SELECT '' AS to_char_9,  to_char(q2, '0999999999999999')       FROM INT8_TBL;
+-- SELECT '' AS to_char_10, to_char(q2, 'S0999999999999999')      FROM INT8_TBL;
+-- SELECT '' AS to_char_11, to_char(q2, 'FM0999999999999999')     FROM INT8_TBL;
+-- SELECT '' AS to_char_12, to_char(q2, 'FM9999999999999999.000') FROM INT8_TBL;
+-- SELECT '' AS to_char_13, to_char(q2, 'L9999999999999999.000')  FROM INT8_TBL;
+-- SELECT '' AS to_char_14, to_char(q2, 'FM9999999999999999.999') FROM INT8_TBL;
+-- SELECT '' AS to_char_15, to_char(q2, 'S 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 . 9 9 9') FROM INT8_TBL;
+-- SELECT '' AS to_char_16, to_char(q2, E'99999 "text" 9999 "9999" 999 "\\"text between quote marks\\"" 9999') FROM INT8_TBL;
+-- SELECT '' AS to_char_17, to_char(q2, '999999SG9999999999')     FROM INT8_TBL;
+
+-- [SPARK-28024] Incorrect value when out of range
+-- check min/max values and overflow behavior
+
+select bigint('-9223372036854775808');
+select bigint('-9223372036854775809');
+select bigint('9223372036854775807');
+select bigint('9223372036854775808');
+
+select bigint('9223372036854775808');
+
+select -(bigint('-9223372036854775807'));
+select -(bigint('-9223372036854775808'));
+
+select bigint('9223372036854775800') + bigint('9223372036854775800');
+select bigint('-9223372036854775800') + bigint('-9223372036854775800');
+
+select bigint('9223372036854775800') - bigint('-9223372036854775800');
+select bigint('-9223372036854775800') - bigint('9223372036854775800');
+
+select bigint('9223372036854775800') * bigint('9223372036854775800');
+
+select bigint('9223372036854775800') / bigint('0');
+select bigint('9223372036854775800') % bigint('0');
+
+select abs(bigint('-9223372036854775808'));
+
+select bigint('9223372036854775800') + int('100');
+select bigint('-9223372036854775800') - int('100');
+select bigint('9223372036854775800') * int('100');
+
+select int('100') + bigint('9223372036854775800');
+select int('-100') - bigint('9223372036854775800');
+select int('100') * bigint('9223372036854775800');
+
+select bigint('9223372036854775800') + smallint('100');
+select bigint('-9223372036854775800') - smallint('100');
+select bigint('9223372036854775800') * smallint('100');
+select bigint('-9223372036854775808') / smallint('0');
+
+select smallint('100') + bigint('9223372036854775800');
+select smallint('-100') - bigint('9223372036854775800');
+select smallint('100') * bigint('9223372036854775800');
+select smallint('100') / bigint('0');
```

Comment:
Please comment out from line 133 to line 173 because all of them expects errors `ERROR:  bigint out of range`.

The only exceptions are line 152, 168, and 173. We can keep those three because Spark returns `NULL` while PostgreSQL expects `ERROR:  division by zero` for those three.